### PR TITLE
Fix CalDAV update if sync is set

### PR DESCRIPTION
**Pull request by bertrandda**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [x] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to [the community](https://community.gladysassistant.com/) for testing before merging.

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

If CalDAV calendar is updated while sync, do not updated sync attribute

### CODE REVIEW

File: server/services/caldav/lib/calendar/calendar.syncUserCalendars.js [hunk 1]
```diff
@@ -53,7 +53,10 @@ async function syncUserCalendars(userId) {
 
       // Else update it if sync is enable on calendar & events change
       if (gladysCalendar[0].sync && formatedCalendar.ctag !== gladysCalendar[0].ctag) {
-        await this.gladys.calendar.update(gladysCalendar[0].selector, formatedCalendar);
```
Comment (Pierre-Gilles) [inline server/services/caldav/lib/calendar/calendar.syncUserCalendars.js:56]: Why not simply:

```
delete formatedCalendar.sync; 
```
 ?

(just asking)

File: server/services/caldav/lib/calendar/calendar.syncUserCalendars.js [hunk 2]
```diff
@@ -53,7 +53,10 @@ async function syncUserCalendars(userId) {
...
+        await this.gladys.calendar.update(gladysCalendar[0].selector, {
+          ...formatedCalendar,
+          sync: gladysCalendar[0].sync,
```
Comment (Pierre-Gilles) [inline server/services/caldav/lib/calendar/calendar.syncUserCalendars.js]: Don't you risk to still update it wrongly if it changed between the .get and the .update ?

Why not simply removing the sync attribute from the update if it's not required?

Comment (bertrandda) [reply server/services/caldav/lib/calendar/calendar.syncUserCalendars.js]: There is no action between get and update, so time is really, really short, I don't think change can happen. But indeed delete is a better option, more readable I changed.

**Review by Pierre-Gilles — APPROVED** [review]
Looks better now! Thanks for the change

**Pull request merged**